File: dietai/src/main/java/com/dietai/dietai/model/Training.java
package com.dietai.dietai.model;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Entity
@Data
@Table(name = "training")
public class Training {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idTraining", columnDefinition = "tinyint")
    private Long idTraining;

    @Column(name = "name", columnDefinition = "Varchar(100)" ,nullable = false, unique = true)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(name = "typeTraining", columnDefinition = "Varchar(20)")
    private TypeTraining typeTraining;

    @Column(name = "difficulty", columnDefinition = "int")
    private Integer difficulty;

    @Column(name = "days", columnDefinition = "int")
    private Integer days;

    @Column(name = "distribution", columnDefinition = "Varchar(50)")
    private String distribution;

    @OneToMany(mappedBy = "training")
    private List<Client> clients;

    @OneToMany(mappedBy = "training")
    private List<TrainingExercise> trainingExercises;

}

File: DietAI-frontend/src/app/models/Training.ts
import { User } from "./User";
import { Client } from "./Client";
import { TypeTraining } from "./Exercise";
import { Exercise } from "./Exercise";

export class Training {
    idTraining: number=0;
    name: string = '';
    typeTraining: string = '';
    difficulty: number=1;
    days: number=1;
    distribution: string='';
    clients?: Client[]=[]; 
}

export class TrainingRequest{
    training: Training = new Training();
}

export class TrainingResponse{
    sets: number=1;
    reps: number=1;
    orderDay: number=1;
    orderWeek: number=1;
    exercise: Exercise = new Exercise();
}

